### ISSUE TITLE
Assertion failure when generating text from LM with Adaptive softmax

### ISSUE THREAD

**Issue by amrsharaf** [pytorch/fairseq#212]
I'm updating the `generate.py` to generate text from a language model pre-trained with an adaptive softmax function. Generation crashes with an assertion failure in the line below:

https://github.com/pytorch/fairseq/blob/1d79ed9b5f67a51e468d012b2ba1bded6d8ebbaf/fairseq/models/fconv.py#L497

I checked the adaptive softmax code and it's seems the targets are only needed for the forward path, is it safe to ignore the assertion and set the `sample[target]` to `None` in the `get_normalized_probs` function?

**Comment by alexeib** [comment]
targets are needed during training and when computing perplexities to make it faster. When you provide targets, you will only get probabilities back for the targets themselves and not anything else.

for generation you want the probability distribution over all toens (presumably). the easiest way to get that from adaptive softmax is, as you suggest, to have target in the sample set to 'None' which tells adaptive softmax that there are no specific targets. 

also if you get generation implemented and working, pull requests are welcome :)

**Comment by amrsharaf** [comment]
Thank you so much. I'll submit a pull request as soon as I finish this.
